### PR TITLE
[Choreo] Code refactoring and base image version upgrades

### PR DESCRIPTION
**Pull request by slahirucd7**
### Purpose
$subject

### Issues

Fixes #https://github.com/wso2-enterprise/choreo/issues/26980

### Automation tests
 - Unit tests added: Yes/No
 - Integration tests added: Yes/No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

File: adapter/internal/messaging/listener.go [hunk 1]
```diff
@@ -32,6 +32,5 @@ func ProcessEvents(config *config.Config) {
 
 	go handleNotification()
 	go handleKMConfiguration()
-	go handleThrottleData()
-	go handleTokenRevocation()
```
Comment (renuka-fernando) [inline adapter/internal/messaging/listener.go:36]: Are we not handling token revocation with the gateway?

**Review by renuka-fernando — APPROVED** [review]

**Review by renuka-fernando — DISMISSED** [review]

File: adapter/internal/adapter/adapter.go [hunk 2]
```diff
@@ -208,7 +208,11 @@ func Run(conf *config.Config) {
 		logger.LoggerMgw.Error("Error reading the log configs. ", errC)
 	}
 
-	logger.LoggerMgw.Info("Starting adapter ....")
+	if conf.GlobalAdapter.Enabled {
+		logger.LoggerMgw.Infof("Preparing adapter in partition: %s to obtain dynamic configurations via xDS....", conf.GlobalAdapter.LocalLabel)
+	} else {
+		logger.LoggerMgw.Info("Preparing adapter to obtain dynamic configurations via xDS....")
```
Comment (VirajSalaka) [inline adapter/internal/adapter/adapter.go:214]: obtain -> serve ?

Comment (slahirucd7) [reply adapter/internal/adapter/adapter.go:214]: +1

**Review by VirajSalaka — APPROVED** [review]

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(NorthEU) cluster : dev-deployment-v2 : 20240222.15

**Comment by choreo-cicd** [comment]
[succeeded] Controlplane cluster : dev-deployment-v2 : 20240222.15

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(EastUS) cluster : dev-deployment-v2 : 20240222.15

**Comment by choreo-cicd** [comment]
[succeeded]  : stage-deployment-v2 : 20240223.1

**Comment by choreo-cicd** [comment]
[failed] Dataplane(EastUS) cluster : stage-deployment-v2 : 20240223.1

**Comment by choreo-cicd** [comment]
[failed] Controlplane cluster : stage-deployment-v2 : 20240223.1

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(NorthEU) cluster : stage-deployment-v2 : 20240223.1

**Comment by choreo-cicd** [comment]
[] Dataplane(NorthEU) cluster : stage-deployment-v2 : 20240224.1

**Comment by choreo-cicd** [comment]
[]  : stage-deployment-v2 : 20240224.1

**Comment by choreo-cicd** [comment]
[] Dataplane(EastUS) cluster : stage-deployment-v2 : 20240224.1

**Comment by choreo-cicd** [comment]
[] Controlplane cluster : stage-deployment-v2 : 20240224.1

**Comment by choreo-cicd** [comment]
[failed] Controlplane cluster : prod-deployment-v2 : 20240227.2

**Comment by choreo-cicd** [comment]
[succeeded] Controlplane cluster : prod-deployment-v2 : 20240227.2

**Comment by choreo-cicd** [comment]
[succeeded]  : prod-deployment-v2 : 20240227.2

**Comment by choreo-cicd** [comment]
[] Dataplane(NorthEU) cluster : prod-deployment-v2 : 20240227.5

**Comment by choreo-cicd** [comment]
[] Dataplane(EastUS) cluster : prod-deployment-v2 : 20240227.5

**Comment by choreo-cicd** [comment]
[]  : prod-deployment-v2 : 20240227.5

**Comment by choreo-cicd** [comment]
[] Controlplane cluster : prod-deployment-v2 : 20240227.5

**Pull request merged**